plugins/versioning/src/main/kotlin/versioning/VersioningConfiguration.kt

```diff
@@ -6,14 +6,23 @@ import java.io.File
 
 data class VersioningConfiguration(
     var olderVersionsDir: File? = defaultOlderVersionsDir,
+    var olderVersions: List<File>? = defaultOlderVersions,
     var versionsOrdering: List<String>? = defaultVersionsOrdering,
     var version: String? = defaultVersion,
 ) : ConfigurableBlock {
     internal fun versionFromConfigurationOrModule(dokkaContext: DokkaContext): String =
         version ?: dokkaContext.configuration.moduleVersion ?: "1.0"
 
+    internal fun allOlderVersions(): List<File> {
+        if (olderVersionsDir != null)
+            assert(olderVersionsDir!!.isDirectory) { "Supplied previous version $olderVersionsDir is not a directory!" }
```

Comment:
Are kotlin's smartcasts not working here?

Reply:
It's since it's a `var`.  I could make it a `?.let` if it's an issue.